File: PA2/app/Models/Product.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Support\Facades\Storage;

class Product extends Model
{
    use HasFactory;

    protected $fillable = [
        'title',
        'image',
        'description',
        'status',
        'price',
        'category_id',
        'user_id', // jika produk dibuat oleh user tertentu
    ];

    protected $casts = [
        'price' => 'float',
    ];

    // Append attribute tambahan ke model JSON output
    protected $appends = ['image_url'];

    /**
     * Relasi ke kategori produk.
     */
    public function category()
    {
        return $this->belongsTo(Category::class);
    }

    /**
     * Relasi ke user (opsional, jika ada creator produk).
     */
    public function user()
    {
        return $this->belongsTo(User::class);
    }

    /**
     * Akses custom: URL gambar produk.
     */
    public function getImageUrlAttribute()
    {
        return $this->image
            ? Storage::url($this->image)
            : asset('images/default.png');
    }
}

File: API/models/product.go
package models

import "time"

type Product struct {
	ID          uint      `json:"id" gorm:"primaryKey"`
	Name        string    `json:"name" gorm:"not null"`
	Description string    `json:"description"`
	Status      string    `json:"status" gorm:"not null"`
	Title       string    `json:"title" gorm:"not null"`
	Price       float64   `json:"price" gorm:"not null"`
	Image       string    `json:"image" gorm:"not null"`
	CreatedAt   time.Time `json:"created_at" gorm:"autoCreateTime"`
	UpdatedAt   time.Time `json:"updated_at" gorm:"autoUpdateTime"`

	// Foreign Key
	CategoryID uint     `json:"category_id"`
	Category   Category `json:"category" gorm:"foreignKey:CategoryID;references:ID"`
}
